### PR TITLE
Use the file explorer view to move between different files

### PR DESCRIPTION
**Pull request by hbjORbj**
2nd part of https://linear.app/intuita/issue/INT-872/show-all-files-in-the-diff-view-and-use-the-file-explorer-view-to-move

https://app.claap.io/intuita/scroll-to-specific-file-on-select-c-BT2DRbCjzO-b01VP7wIPW5R

### CODE REVIEW

File: intuita-webview/src/fileExplorer/TreeView/index.tsx [hunk 1]
```diff
@@ -145,6 +149,17 @@ const TreeView = ({ node, nodeIds, fileNodes, searchQuery }: Props) => {
 		};
 	}, []);
 
+	useEffect(() => {
+		if (focusedNodeId === null || !fileNodeIds.has(focusedNodeId)) {
+			return;
+		}
+
+		vscode.postMessage({
+			kind: 'webview.fileExplorer.selectFile',
```
Comment (gregintuitaio) [inline intuita-webview/src/fileExplorer/TreeView/index.tsx:158]: is it a command (`selectFile`) or an event (`fileSelected`)?

File: src/components/webview/FileExplorerProvider.ts [hunk 2]
```diff
@@ -38,14 +39,15 @@ import {
...
+import { DiffWebviewPanel } from './DiffWebviewPanel';
 
 export class FileExplorerProvider implements WebviewViewProvider {
 	__view: WebviewView | null = null;
 	__extensionPath: Uri;
 	__webviewResolver: WebviewResolver | null = null;
 	__elementMap = new Map<ElementHash, Element>();
 	__folderMap = new Map<string, TreeNode>();
-	__fileNodes = new Set<TreeNode>();
+	__fileNodes = new Map<string, { jobHash: JobHash; node: TreeNode }>();
```
Comment (gregintuitaio) [inline src/components/webview/FileExplorerProvider.ts:50]: is this a map between URIs to jobs?

File: src/components/webview/FileExplorerProvider.ts [hunk 3]
```diff
@@ -394,6 +405,35 @@ export class FileExplorerProvider implements WebviewViewProvider {
 				...(message.value.arguments ?? []),
 			);
 		}
+
+		if (message.kind === 'webview.fileExplorer.selectFile' && message.id) {
```
Comment (gregintuitaio) [inline src/components/webview/FileExplorerProvider.ts:409]: how would a `message.id` be not set?

File: src/components/webview/webviewEvents.ts [hunk 4]
```diff
@@ -203,6 +207,10 @@ export type WebviewResponse =
 			kind: 'webview.campaignManager.selectCase';
 			hash: string | null;
 	  }>
+	| Readonly<{
+			kind: 'webview.fileExplorer.selectFile';
+			id: string | null;
```
Comment (gregintuitaio) [inline src/components/webview/webviewEvents.ts:212]: should be null?

File: intuita-webview/src/fileExplorer/TreeView/index.tsx (hunk 1 above)
Comment (hbjORbj) [reply intuita-webview/src/fileExplorer/TreeView/index.tsx:158]: Event. `fileSelected`. Thanks. Changed it.

File: src/components/webview/FileExplorerProvider.ts (hunk 3 above)
Comment (hbjORbj) [reply src/components/webview/FileExplorerProvider.ts:409]: addressed

File: src/components/webview/FileExplorerProvider.ts (hunk 2 above)
Comment (hbjORbj) [reply src/components/webview/FileExplorerProvider.ts:50]: It's a map between URIs to the File Tree Node and the job hash.

Left it as a comment

**Review by grzpab — APPROVED** [review]

**Pull request merged**